ludwig/backend/ray.py

```diff
@@ -176,16 +177,20 @@ def batch_predict(self, model, dataset, *args, **kwargs):
         predictor_kwargs = self.predictor_kwargs
         output_columns = get_output_columns(model.output_features)
 
-        def batch_predict_partition(dataset):
+        def batch_predict_partition(df):
+            pd_ds = PandasDataset(df, dataset.features, dataset.data_hdf5_fp)
             model = remote_model.load()
             predictor = Predictor(**predictor_kwargs)
-            predictions = predictor.batch_predict(model, dataset, *args, **kwargs)
+            predictions = predictor.batch_predict(model, pd_ds, *args, **kwargs)
             ordered_predictions = predictions[output_columns]
             return ordered_predictions
-
-        return dataset.map_dataset_partitions(
-            batch_predict_partition,
-            meta=[(c, 'object') for c in output_columns]
+        
+        num_gpus = int(ray.cluster_resources().get('GPU', 0) > 0)
+        return dataset.ds.map_batches(
```

Comment:
We may need to do a conversion to Dask DataFrame here:

```
dataset.ds.map_batches(...).to_dask()
```

The other option is to carry through the RayDataset and then only call the methods to save to a particular format in the end. This might be a better longterm solution, but might be too involved for this PR, since the code already assumes Dask DF will be used downstream. What do you think?

Reply:
Yeah I agree, I really like the idea of carrying Ray Datasets throughout the prediction and I can do that in a new PR.